lib/mix/lib/mix/dep/loader.ex

```diff
@@ -301,17 +304,18 @@ defmodule Mix.Dep.Loader do
     |> elem(0)
   end
 
-  defp rebar_children(app, root_config, extra, manager) do
+  defp rebar_children(_app, root_config, manager) do
     from = Path.absname("rebar.config")
     Mix.Rebar.recur(root_config, fn config ->
-      app
-      |> Mix.Rebar.deps(config, overrides(manager, extra))
```

Comment:
@ericmj The previous version was applying overrides ONLY for dependencies. This PR changes it and we apply overrides for any config in the child configuration.

That would probably be a good test to add, which this PR is messing. If you can, please push a test to this PR. :heart: